api/routers/edr.py

```diff
@@ -89,13 +97,24 @@ async def get_data_location_id(
 )
 async def get_data_position(
     coords: str = Query(..., example="POINT(5.179705 52.0988218)"),
-    parameter_name: str = Query(..., alias="parameter-name", example="dd,ff,rh,pp,tn"),
-    datetime: str | None = None,
+    parameter_name: Annotated[str | None, Query(alias="parameter-name", example="dd,ff,rh,pp,tn")] = None,
+    datetime: Annotated[str | None, Query(alias="datetime", example="2022-12-31T00:00Z/2023-01-01T00:00Z")] = None,
     f: str = Query(default="covjson", alias="f", description="Specify return format."),
 ):
-    point = wkt.loads(coords)
-    assert point.geom_type == "Point"
-    poly = buffer(point, 0.0001, quad_segs=1)  # Roughly 10 meters around the point
+    try:
+        point = wkt.loads(coords)
+        if point.geom_type != "Point":
+            raise TypeError
+        poly = buffer(point, 0.0001, quad_segs=1)  # Roughly 10 meters around the point
+    except GEOSException:
+        raise HTTPException(status_code=400, detail={"coords": f"Invalid or unparseable wkt provided: {coords}"})
+    except TypeError:
+        raise HTTPException(status_code=400, detail={"coords": f"Invalid geometric type: {point.geom_type}"})
```

Comment:
Why not just raise the `HTTPException` directly on line 107?

Reply:
The general `Exception` would catch it and then raise a new `HTTPException` with incorrect details. The general `Exception` was left there as a safety measure because even though shapely uses `GEOSExpection` as a base class, `ValueError`  can be raised in some cases.

Reply:
Hmm, I don't like communication by Exception over a couple of lines, but I see the point now. Another option would be to just reraise the HTTP exception in the final handler, but not sure that is much nicer...